src/kbc_runtime.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::*;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use super::kbc_modules;
+
+pub type KbcInstance = Box<dyn KbcInterface + Sync + Send>;
+
+// KbcInterface is a standard interface that all KBC modules need to implement.
+pub trait KbcInterface {
+    fn check(&self) -> KbcCheckInfo;
+    fn decrypt_payload(&mut self, annotation: &str) -> Result<Vec<u8>>;
+}
+
+// KbcCheckInfo is used by KBC module instances to report their internal status to AA.
+pub struct KbcCheckInfo {
+    pub kbs_info: Vec<String>,
```

Comment:
Is `Vec<String>` really the best way to store kbs_info? Seems like maybe we want some kind of dictionary to store different info fields?